Assets/HoloToolkit/SpatialUnderstanding/Scripts/SpatialUnderstandingCustomMesh.cs

```diff
@@ -118,6 +123,8 @@ public void Commit()
                     MeshObject.SetTriangles(tris, 0);
                     MeshObject.RecalculateNormals();
                     MeshObject.RecalculateBounds();
+                    SpatialCollider.sharedMesh = null;
+                    SpatialCollider.sharedMesh = MeshObject;
```

Comment:
Primitive colliders are not feasible -- mesh colliders are used for spatial mapping. That said, IIRC, the code that generates these meshes runs "asynchronously" as a coroutine and periodically yields.  Also, it appears to me that Spatial Understanding produces a lot of really tiny meshes.  

If you look at where Commit() is called, on line 319, you can see that the routine yields immediately afterwards. This should be fine, especially considering that this is performed only once when finalizing the mesh.